red-hat-data-services/odh-dashboard
# [UPSTREAM] Refactor fetching groups (#242)

- [X] JIRA link(s): https://issues.redhat.com/browse/RHODS-3549

## Live build

quay.io/lferrnan/rhods-operator-live-catalog:1.12.1-admin-group

## Testing Instructions

1. Install the live build
2. Check that there is no `Failed to get groups: TypeError: Cannot read property 'includes' of null` message in the fresh install.
3. If the group is not created, a new desriptive error would trigger to indicate wich group should you create. `Error: Failed to retrieve Group dedicated-admins, might not exist.`
4. Create that group to dismiss the error.
5. Check that if you remove `groups-config` ConfigMap you will have a descriptive error. `Error: Failed to retrieve ConfigMap groups-config, might be malformed or doesn't exist.`
6. If you add it again, and in the second configmap (usually rhods-groups-config) you change the `admin_groups` field to `'system:authenticated'` you will have a descriptive error. `Error: It is not allowed to set "system:authenticated" or an empty string as admin group`
